engine/src/flutter/lib/ui/hooks.dart

```diff
@@ -307,6 +307,16 @@ void _dispatchPointerDataPacket(ByteData packet) {
   PlatformDispatcher.instance._dispatchPointerDataPacket(packet);
 }
 
+@pragma('vm:entry-point')
+bool _platformViewShouldAcceptTouch(int viewId, double x, double y) {
+  Offset offset = Offset(x, y);
+  // Engine must pass in a valid viewId.
+  FlutterView view = PlatformDispatcher.instance._views[viewId]!;
```

Comment:
leftover from our discussion @chunhtai & @loic-sharma. Feel free to continue discussion here.

Reply:
This is indeed the correct thing for us to do. We have the same check to verify viewport metrics are sent to a valid view:

https://github.com/flutter/flutter/blob/a7e5fa1da6eaee5cbb3df709f5581cb5926b7b27/engine/src/flutter/lib/ui/platform_dispatcher.dart#L332-L336